### ISSUE TITLE
Mobile app: Improve current `develop` branch

### ISSUE THREAD

**Issue by gianalarcon** [Quantum3-Labs/themarquis-app#16]
## Issue Overview

- There has been many PRs that got merged, need to re-conciliate them with main branch

## Proposed Solutions or Ideas

- Clean, integrate and test whole gameplay.
- Check missing relevant edge cases.

**Comment by wms2537** [comment]
Can I jump on this task?

**Comment by co01y4p** [comment]
@gianalarcon @wms2537 
Helo! Came across to inform the version has following issues:
1. The loading indicator is on top/ inside the cyan blue button, which it should replace the button, rather than overlap it

**Comment by gianalarcon** [comment]
Thanks for reporting, we can add it as part of the scope of this issue #16

> @gianalarcon @wms2537 Helo! Came across to inform the version has following issues:
> 
> 1. The loading indicator is on top/ inside the cyan blue button, which it should replace the button, rather than overlap it

**Comment by co01y4p** [comment]
2. The appbar in game screen is missing.
3. The wording of "No tokens available..... start!" is too small to read.

![Screenshot 2024-12-03 at 00 50 49](https://github.com/user-attachments/assets/14d6e97d-b3fa-47ac-872f-bd3fc3bc17c4)

**Comment by wms2537** [comment]
The waiting room not updated when user join (current user changes not reflected). The bottom start button is changing, but the last user box pic is still empty

**Comment by wms2537** [comment]
The menu screen background display got problem, seems the overlay scaling problem. It is displaying a enlarged version of the board when I return from the game to the Game Menu Screen.

**Comment by wms2537** [comment]
The dice rolling animation, after roll dice no stop and show the rolled value

**Comment by wms2537** [comment]
```
he following CircularDependencyError was thrown while handling a gesture:
Instance of 'CircularDependencyError'

When the exception was thrown, this was the stack:
#0      ProviderElementBase._debugAssertCanDependOn.<anonymous closure> (package:riverpod/src/framework/element.dart:645:13)
element.dart:645
#1      ProviderElementBase._debugAssertCanDependOn (package:riverpod/src/framework/element.dart:650:8)
element.dart:650
#2      ProviderElementBase.read (package:riverpod/src/framework/element.dart:674:12)
element.dart:674
#3      AppState.selectGame (package:marquis_v2/providers/app_state.dart:50:9)
app_state.dart:50
#4      _HomeScreenState.build.<anonymous closure> (package:marquis_v2/screens/home_screen.dart:105:40)
home_screen.dart:105
#5      _InkResponseState.handleTap (package:flutter/src/material/ink_well.dart:1176:21)
ink_well.dart:1176
#6      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:351:24)
recognizer.dart:351
#7      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:656:11)
tap.dart:656
#8      BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:313:5)
tap.dart:313
#9      BaseTapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:246:7)
tap.dart:246
#10     PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:703:9)
recognizer.dart:703
#11     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:98:12)
pointer_router.dart:98
#12     PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:143:9)
pointer_router.dart:143
#13     _LinkedHashMapMixin.forEach (dart:_compact_hash:726:13)
dart:_compact_hash:726
#14     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:141:18)
pointer_router.dart:141
#15     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:127:7)
pointer_router.dart:127
#16     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:502:19)
binding.dart:502
#17     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:482:22)
binding.dart:482
#18     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:457:11)
binding.dart:457
#19     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:427:7)
binding.dart:427
#20     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:390:5)
binding.dart:390
#21     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:337:7)
binding.dart:337
#22     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:306:9)
binding.dart:306
#23     _invoke1 (dart:ui/hooks.dart:328:13)
hooks.dart:328
#24     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:442:7)
platform_dispatcher.dart:442
#25     _dispatchPointerDataPacket (dart:ui/hooks.dart:262:31)
hooks.dart:262

Handler: "onTap"
Recognizer: TapGestureRecognizer#35c71
    debugOwner: GestureDetector
    state: possible
    won arena
    finalPosition: Offset(391.0, 409.3)
    finalLocalPosition: Offset(27.0, 12.6)
    button: 1
    sent tap down
   ```

**Comment by wms2537** [comment]
> ```
> he following CircularDependencyError was thrown while handling a gesture:
> Instance of 'CircularDependencyError'
> 
> When the exception was thrown, this was the stack:
> #0      ProviderElementBase._debugAssertCanDependOn.<anonymous closure> (package:riverpod/src/framework/element.dart:645:13)
> element.dart:645
> #1      ProviderElementBase._debugAssertCanDependOn (package:riverpod/src/framework/element.dart:650:8)
> element.dart:650
> #2      ProviderElementBase.read (package:riverpod/src/framework/element.dart:674:12)
> element.dart:674
> #3      AppState.selectGame (package:marquis_v2/providers/app_state.dart:50:9)
> app_state.dart:50
> #4      _HomeScreenState.build.<anonymous closure> (package:marquis_v2/screens/home_screen.dart:105:40)
> home_screen.dart:105
> #5      _InkResponseState.handleTap (package:flutter/src/material/ink_well.dart:1176:21)
> ink_well.dart:1176
> #6      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:351:24)
> recognizer.dart:351
> #7      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:656:11)
> tap.dart:656
> #8      BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:313:5)
> tap.dart:313
> #9      BaseTapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:246:7)
> tap.dart:246
> #10     PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:703:9)
> recognizer.dart:703
> #11     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:98:12)
> pointer_router.dart:98
> #12     PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:143:9)
> pointer_router.dart:143
> #13     _LinkedHashMapMixin.forEach (dart:_compact_hash:726:13)
> dart:_compact_hash:726
> #14     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:141:18)
> pointer_router.dart:141
> #15     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:127:7)
> pointer_router.dart:127
> #16     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:502:19)
> binding.dart:502
> #17     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:482:22)
> binding.dart:482
> #18     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:457:11)
> binding.dart:457
> #19     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:427:7)
> binding.dart:427
> #20     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:390:5)
> binding.dart:390
> #21     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:337:7)
> binding.dart:337
> #22     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:306:9)
> binding.dart:306
> #23     _invoke1 (dart:ui/hooks.dart:328:13)
> hooks.dart:328
> #24     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:442:7)
> platform_dispatcher.dart:442
> #25     _dispatchPointerDataPacket (dart:ui/hooks.dart:262:31)
> hooks.dart:262
> 
> Handler: "onTap"
> Recognizer: TapGestureRecognizer#35c71
>     debugOwner: GestureDetector
>     state: possible
>     won arena
>     finalPosition: Offset(391.0, 409.3)
>     finalLocalPosition: Offset(27.0, 12.6)
>     button: 1
>     sent tap down
> ```

Fixed in #17
